app/code/core/Mage/Eav/Model/Resource/Entity/Attribute.php

```diff
@@ -521,7 +521,7 @@ protected function _afterLoad(Mage_Core_Model_Abstract $object)
                 $object->addData($result);
             }
         }
-
+        $object->setDataChanges(false);
```

Comment:
This change should preferably be done in `Mage_Eav_Model_Entity_Attribute_Abstract::loadByCode`. You may also `setOrigData` since it seems it was also forgotten.

```diff
    public function loadByCode($entityType, $code)
    {
        Varien_Profiler::start('_LOAD_ATTRIBUTE_BY_CODE__');
        if (is_numeric($entityType)) {
            $entityTypeId = $entityType;
        } elseif (is_string($entityType)) {
            $entityType = Mage::getModel('eav/entity_type')->loadByCode($entityType);
        }
        if ($entityType instanceof Mage_Eav_Model_Entity_Type) {
            $entityTypeId = $entityType->getId();
        }
        if (empty($entityTypeId)) {
            throw Mage::exception('Mage_Eav', Mage::helper('eav')->__('Invalid entity supplied.'));
        }
        $this->_getResource()->loadByCode($this, $entityTypeId, $code);
        $this->_afterLoad();
+       $this->setOrigData(); 
+       $this->_hasDataChanges = false;
        Varien_Profiler::stop('_LOAD_ATTRIBUTE_BY_CODE__');
        return $this;
    }
```